src/node/historical_queries_adapter.cpp

```diff
@@ -190,6 +190,57 @@ namespace ccf::historical
     return tx_id_opt;
   }
 
+  void default_error_handler(
+    HistoricalQueryErrorCode err,
+    std::string reason,
+    endpoints::CommandEndpointContext& args)
+  {
+    if (err == HistoricalQueryErrorCode::InternalError)
```

Comment:
switch?

Reply:
It requires all variable declaration to be moved out of the whole `switch` block, and to me personally looks less readable for inner blocks with multiple function calls inside.

It also might grow more for new error codes in the future, as well as more `ctx` calls in each block, which will make this block grow even further.

That's why I changed that to `if`-`else`.

Let me know if you see a good reason to go back to `switch`.

Reply:
I don't understand, there are no declarations here, and even if there were, it's no more verbose to have blocks under each case as it is with an if-else. It's still one block per case?

```
switch (err)
{
 case FirstErr:
 {
   auto a = foo();
   break;
 }
...
}
```

Aside from performance, switch gives us better static checking with our compiler settings, missing cases will be flagged. This in my opinion outweighs syntactic concerns, particularly if we intend to grow the enum.

Reply:
Agree for combining error for one case.

Disagree to compiler warnings because I tried to comment out some cases, compiler doesn't notice.

Anyway, changing to switch as the majority prefers it.